File: src/types/thread.ts
export interface Thread {
  id: string;
  sessionId: string;
  forkedFromId?: string;
  parentThreadId?: string;
  preview: string;
  name?: string;
  status: ThreadStatus;
  createdAt: number;
  updatedAt: number;
  archived: boolean;
}

export type ThreadStatus = "idle" | "active" | "loading" | "error";

export interface ThreadStartParams {
  model?: string;
  modelProvider?: string;
  cwd?: string;
  approvalPolicy?: ApprovalPolicy;
  instructions?: string;
}

export interface ThreadStartResponse {
  thread: Thread;
  model: string;
  modelProvider: string;
  serviceTier?: string;
  cwd: string;
}

export interface ThreadResumeParams {
  threadId: string;
}

export interface ThreadResumeResponse {
  thread: Thread;
}

export interface ThreadListParams {
  cursor?: string;
  limit?: number;
  archived?: boolean;
}

export interface ThreadListResponse {
  data: Thread[];
  nextCursor?: string;
}

export interface ThreadReadParams {
  threadId: string;
  includeTurns?: boolean;
}

export interface ThreadReadResponse {
  thread: Thread;
  turns?: Turn[];
}

export interface ThreadArchiveParams {
  threadId: string;
}

export interface ThreadArchiveResponse {}

export interface ThreadUnarchiveParams {
  threadId: string;
}

export interface ThreadUnarchiveResponse {
  thread: Thread;
}

export interface ThreadSetNameParams {
  threadId: string;
  name: string;
}

export interface ThreadSetNameResponse {}

export interface ThreadRollbackParams {
  threadId: string;
  dropCount: number;
}

export interface ThreadRollbackResponse {
  thread: Thread;
}

export interface ThreadUnsubscribeParams {
  threadId: string;
}

export interface ThreadUnsubscribeResponse {}

export type ApprovalPolicy =
  | "untrusted"
  | "on-failure"
  | "on-request"
  | "never";

import type { Turn } from "./turn";

File: src-tauri/src/commands/thread.rs
use tauri::State;

use crate::error::{AppError, AppResult};
use crate::protocol::*;
use crate::state::AppState;

/// 这些命令是旧 codex 桥接模式的遗留接口。
/// 在 standalone 模式下，前端使用 standalone_* 命令，这些命令不会被调用。
/// 保留注册以防前端有残留调用时给出明确错误提示。
fn not_supported() -> AppError {
    AppError::Custom("This command is not available in standalone mode".to_string())
}

#[tauri::command]
pub async fn thread_start(
    state: State<'_, AppState>,
    params: ThreadStartParams,
) -> AppResult<ThreadStartResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_resume(
    state: State<'_, AppState>,
    params: ThreadResumeParams,
) -> AppResult<ThreadResumeResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_list(
    state: State<'_, AppState>,
    params: ThreadListParams,
) -> AppResult<ThreadListResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_read(
    state: State<'_, AppState>,
    params: ThreadReadParams,
) -> AppResult<ThreadReadResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_archive(
    state: State<'_, AppState>,
    params: ThreadArchiveParams,
) -> AppResult<ThreadArchiveResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_unarchive(
    state: State<'_, AppState>,
    params: ThreadUnarchiveParams,
) -> AppResult<ThreadUnarchiveResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_set_name(
    state: State<'_, AppState>,
    params: ThreadSetNameParams,
) -> AppResult<ThreadSetNameResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_rollback(
    state: State<'_, AppState>,
    params: ThreadRollbackParams,
) -> AppResult<ThreadRollbackResponse> {
    let _ = (state, params);
    Err(not_supported())
}

#[tauri::command]
pub async fn thread_unsubscribe(
    state: State<'_, AppState>,
    params: ThreadUnsubscribeParams,
) -> AppResult<ThreadUnsubscribeResponse> {
    let _ = (state, params);
    Err(not_supported())
}
